app/views/DiscussionsView/index.tsx

```diff
@@ -0,0 +1,230 @@
+		setIsSearching(true);
+		await load(text);
+	}, 300);
+
+	const onCancelSearchPress = () => {
+		setIsSearching(false);
+		setSearch([]);
+		setSearchTotal(0);
+	};
+
+	const onSearchPress = () => {
+		setIsSearching(true);
+	};
+
+	const setHeader = () => {
+		let options: Partial<StackNavigationOptions>;
+		if (isSearching) {
+			const headerTitlePosition = getHeaderTitlePosition({ insets, numIconsRight: 1 });
+			options = {
+				headerTitleAlign: 'left',
+				headerLeft: () => (
+					<HeaderButton.Container left>
+						<HeaderButton.Item iconName='close' onPress={onCancelSearchPress} />
+					</HeaderButton.Container>
+				),
+				headerTitle: () => (
+					<SearchHeader
+						placeholder='Search Messages'
+						onSearchChangeText={onSearchChangeText}
+						theme={theme}
+						testID='discussion-messages-view-search-header'
+					/>
+				),
+				headerTitleContainerStyle: {
+					left: headerTitlePosition.left,
+					right: headerTitlePosition.right
+				},
+				headerRight: () => null
+			};
+			return options;
+		}
+
+		options = {
+			headerLeft: () => (
+				<HeaderBackButton labelVisible={false} onPress={() => navigation.pop()} tintColor={themes[theme!].headerTintColor} />
+			),
+			headerTitleAlign: 'center',
+			headerTitle: I18n.t('Discussions'),
+			headerTitleContainerStyle: {
+				left: null,
+				right: null
+			},
+			headerRight: () => (
+				<HeaderButton.Container>
+					<HeaderButton.Item iconName='search' onPress={onSearchPress} />
+				</HeaderButton.Container>
+			)
+		};
+
+		if (isMasterDetail) {
+			options.headerLeft = () => <HeaderButton.CloseModal navigation={navigation} />;
+		}
+		return options;
+	};
+
+	useEffect(() => {
+		load();
+	}, []);
+
+	useEffect(() => {
+		const options = setHeader();
+		navigation.setOptions(options);
+	}, [navigation, isSearching]);
+
+	const onDiscussionPress = debounce(
+		(item: any) => {
+			navigation.push('RoomView', {
+				rid: item.drid,
+				prid: item.rid,
+				name: item.msg,
+				t: item.rid === 'GENERAL' ? 'c' : 'p'
+			});
+		},
+		1000,
+		true
+	);
+
+	const renderItem = ({ item }: any) => (
+		<Item
+			{...{
+				item,
+				user,
+				navigation,
+				baseUrl,
+				useRealName
+			}}
+			timeFormat={timeFormat}
+			onPress={onDiscussionPress}
+		/>
+	);
+
+	if (!discussions?.length) {
+		return <BackgroundContainer loading={loading} text={I18n.t('No_discussions')} />;
+	}
+
+	return (
+		<SafeAreaView testID='discussions-view'>
+			<StatusBar />
+			<FlatList
+				data={isSearching ? search : discussions}
+				renderItem={renderItem}
+				keyExtractor={(item: any) => item.msg}
+				style={{ backgroundColor: themes[theme!].backgroundColor }}
+				contentContainerStyle={styles.contentContainer}
+				onEndReachedThreshold={0.5}
+				maxToRenderPerBatch={5}
+				windowSize={10}
+				initialNumToRender={7}
+				removeClippedSubviews={isIOS}
+				onEndReached={() => (isSearching ? searchTotal > API_FETCH_COUNT ?? load() : total > API_FETCH_COUNT ?? load())}
```

Comment:
Hey, we need to add https://eslint.org/docs/rules/no-nested-ternary 
I just can't read this, I'm sorry 🙈